File: algorithms/count_sort/javascript/count_sort.js
"use strict";
var countSort = (arr, range)=>{
    /**
     * Consider the input is in range of 1 to 9
      */
    // this can vary depending on your input array
    let countArr = [];
    //Initialise count array to 0;
    for(let i=0; i <= range; i++){
        countArr[i] = 0;
    }
    for(let i=0; i < arr.length; i++){
        ++countArr[arr[i]];
    }
    for(let i=1; i<= countArr.length-1; i++){
        countArr[i] += countArr[i-1];
    }
    //Take the sorted array and insert the data of input array based on the count array and
    // decrease the value by 1 every time the element is inserted in sorted array

    let sortedArray = new Array(arr.length -1);
    for(let i=0; i<arr.length; i++){
        sortedArray[countArr[arr[i]]-1] = arr[i];
        --countArr[arr[i]];
    }
 return sortedArray;    
}
let arr = [2,1,3,4,5,4,6,7,6];
let range = 9;
let sortedArray = countSort(arr, range);
console.log(sortedArray);

File: algorithms/count_sort/java/count_sort.java
	public int[] count_sort(int[] a)
	{
		int[] result = new int[a.length];
		int[] count = new int[a.length];
		for (int i = 0; i < a.length; i++)
		{
			count[i] = 0;
		}
		for (int i = 0; i < a.length-1; i++)
		{
			for (int j = i + 1; j < a.length; j++)
			{
				if (a[i] < a[j])
					count[j] = count[j] + 1;
				else
					count[i] = count[i] + 1;
			}
		}
		for (int i = 0; i < a.length; i++)
		{
			result[count[i]] = a[i];
		}
		return result;
	}